src/stim/circuit/gate_data.h

```diff
@@ -44,7 +44,46 @@ struct ErrorAnalyzer;
 constexpr uint8_t ARG_COUNT_SYGIL_ANY = uint8_t{0xFF};
 constexpr uint8_t ARG_COUNT_SYGIL_ZERO_OR_ONE = uint8_t{0xFE};
 
-inline uint8_t gate_name_to_id(const char *v, size_t n) {
+enum class Gates : uint8_t {
+    // From gate_data_annotations.cc
+    DETECTOR, OBSERVABLE_INCLUDE, TICK, QUBIT_COORDS, SHIFT_COORDS,
+    // Frome gate_data_blocks.cc
+    REPEAT,
+    // From gate_data_collapsing.cc
+    MX, MY, M, MRX, MRY, MR, RX, RY, R, MPP,
+    // From gate_data_controlled.cc
+    XCX, XCY, XCZ, YCX, YCY, YCZ, CX, CY, CZ,
+    // From gate_data_hada.cc
+    H, H_XY, H_YZ,
+    // From gate_data_noisy.cc
+    DEPOLARIZE1, DEPOLARIZE2, X_ERROR, Y_ERROR, Z_ERROR,
+    PAULI_CHANNEL_1, PAULI_CHANNEL_2,
+    E, ELSE_CORRELATED_ERROR,
+    // From gate_data_pauli.cc
+    I, X, Y, Z,
+    // From gate_data_period_3.cc
+    C_XYZ, C_ZYX,
+    // From gate_data_period_4.cc
+    SQRT_X, SQRT_X_DAG, SQRT_Y, SQRT_Y_DAG, S, S_DAG,
+    // From gate_data_pp.cc
+    SQRT_XX, SQRT_XX_DAG, SQRT_YY, SQRT_YY_DAG, SQRT_ZZ, SQRT_ZZ_DAG,
+    // From gate_data_swaps.cc
+    SWAP, ISWAP, CXSWAP, SWAPCX, ISWAP_DAG,
+
+    /// GATE ALIASES
```

Comment:
We shouldn't include the aliases in the enum.

For now we can keep them if it makes the change smoother, but prefix them with `ALIAS_TO_REMOVE_` or something.

Reply:
With the changes since this feedback, I was able to simply remove them from the enum.